enigma-cli/src/main/java/cuchaz/enigma/command/Command.java

```diff
@@ -99,49 +98,14 @@ public static EnigmaProject openProject(Path fileJarIn, Path fileMappings, Enigm
 		return project;
 	}
 
-	protected static EntryTree<EntryMapping> readMappings(Path path, ProgressListener progress, MappingSaveParameters saveParameters) throws Exception {
-		List<Exception> suppressed = new ArrayList<>();
-		if ("zip".equalsIgnoreCase(MoreFiles.getFileExtension(path))) {
-			return MappingFormat.ENIGMA_ZIP.read(path, progress, saveParameters);
-		} else {
-			for (MappingFormat format : MappingFormat.getReadableFormats()) {
-				try {
-					return format.read(path, progress, saveParameters);
-				} catch (Exception e) {
-					suppressed.add(e);
-				}
-			}
-		}
-
-		RuntimeException exception = new RuntimeException("Unable to parse mappings!");
-		for (Exception supressedException : suppressed) {
-			exception.addSuppressed(supressedException);
-		}
-
-		throw exception;
+	protected static EntryTree<EntryMapping> readMappings(Path path, ProgressListener progress, MappingSaveParameters saveParameters) throws MappingParseException, IOException {
+		MappingFormat format = MappingFormat.parseFromFile(path);
+		return format.read(path, progress, saveParameters);
 	}
 
-	protected static void writeMappings(EntryTree<EntryMapping> mappings, Path path, ProgressListener progress, MappingSaveParameters saveParameters) throws Exception {
-		List<Exception> suppressed = new ArrayList<>();
-		if ("zip".equalsIgnoreCase(MoreFiles.getFileExtension(path))) {
-			MappingFormat.ENIGMA_ZIP.write(mappings, path, progress, saveParameters);
-		} else {
-			for (MappingFormat format : MappingFormat.getWritableFormats()) {
-				try {
-					format.write(mappings, path, progress, saveParameters);
-					return;
-				} catch (Exception e) {
-					suppressed.add(e);
-				}
-			}
-		}
-
-		RuntimeException exception = new RuntimeException("Unable to write mappings!");
-		for (Exception supressedException : suppressed) {
-			exception.addSuppressed(supressedException);
-		}
-
-		throw exception;
+	protected static void writeMappings(EntryTree<EntryMapping> mappings, Path path, ProgressListener progress, MappingSaveParameters saveParameters) {
```

Comment:
I think it would be better to entirely remove this method, here `path` might not be an existing file, and the only command that uses it is actually meant to use the same format as its input.